### ISSUE TITLE
Tutorial sobre como escrever uma issue

### ISSUE THREAD

**Issue by fabriciofx** [ifpe-cti/modelo-projeto-ppo#5]
Muitos estudantes encontram problemas ao escrever uma issue, como não especificar o que deve ser feito na issue, escrevê-la muito genérica, ou não marcar as pessoas que vão atuar na issue. Assim, devemos escrever um tutorial esclarecendo estes pontos.

**Comment by mdbs99** [comment]
Uma issue poderá ser escrita tendo diferentes visões. Por exemplo, a **visão técnica** vs. **visão de usuário**. 

De acordo com a minha experiência, o usuário tem a tendência de escrever uma issue muito genérica e sem muitos detalhes. Isso pode ser um problema, pois uma issue genérica pode gerar muitas tarefas específicas, o que irá gerar muitas outras sub-issues. Cabe ao arquiteto e/ou gerente de projetos dividir a issue em partes menores e mais gerenciáveis, afim de distribuir o trabalho para a equipe. Mas, isso pode gerar outro tipo de problema.

Nem o sistema do Github nem nenhum outro (que eu conheça) tem o suporte a sub-issues. Por exemplo, se uma issue A gerar as sub-issues A.1 e A.2, o sistema não gerencia o fechamento automático da issue A quando suas sub-issues forem concluídas. Isso gera um trabalho "manual" para o arquiteto, pois ele tem que saber o que fazer (qual o próximo passo) quando a issue A.2 referente a issue A for concluída.

A visão técnica, ou seja, quando desenvolvedores escrevem suas próprias issues, tem a tendência de ser bem específica e documentada. Resolve apenas um problema e são fácies de gerenciar pois, na verdade, quem gerencia é o próprio desenvolvedor.

Abaixo um exemplo de como seria esse fluxo:

1. O usuário gera a issue A;
2. O arquiteto quebra a issue em A.1 e A.2 e designa a 2 desenvolvedores;
3. O desenvolvedor 1 quebra a issue A.1 em mais 5 issues menores, mas o arquiteto não precisa gerenciá-las (a princípio, se tudo estiver correndo bem);
4. O desenvolvedor 2 faz algo similar;
5. Quando ambos os desenvolvedores fecharem A.1 e A.2, o arquiteto poderá fechar a issue A ou criar novas sub-issues, se for o caso.

**Comment by fabriciofx** [comment]
@mdbs99 Concordo plenamente com tudo que você escreveu. Agradeço *muito* pelos comentários e por compartilhar a sua experiência conosco!

**Comment by fabriciofx** [comment]
Colocar também as razões abaixo do porquê utilizar o login dos usuários do GitHub com labels e com esse recurso, associar uma issue a um (ou mais) usuário/desenvolvedor:

1. Apenas colaboradores de um projeto/repositório podem atribuir (assign) uma issue para outros colaboradores ou colocar labels em uma issue;
2. Eu, sendo colaborador, só posso atribuir uma issue a uma pessoa que também é colaborador do mesmo projeto/repositório. E se eu precisar atribuir aquela issue a uma pessoa que não é colaborador? Eu posso criar um label para ela (com o login dela no GH) e etiquetar a issue, dessa forma ficando claro que aquela issue foi associada a determinado usuário que não necessariamente é colaborador do projeto;
3. Só é possível fazer buscas nas issues por um author por vez. Em compensação, posso fazer buscas por mais de um label por vez. Assim, utilizando o esquema de login nos labels, posso fazer uma busca por issues que foram atribuídas a mais de uma pessoa (mas IMHO isso é um erro de gerenciamento: cada issue só deveria ter **UM** responsável);
4. Visualmente fica melhor de ver e entender quais issues estão associadas a quais pessoas utilizando o esquema de labels nos logins.

@mdbs99 Esse texto é um pequeno resumo daquele conversa que tivemos. Se quiser fazer algum comentário, fique à vontade! 😄

**Comment by viniciussoaresti** [comment]
É necessário também o controle de quem faz a issue de ser conciso, direto, mas não generalizar demais a coisa, ao ponto de ser necessária toda hora uma retomada à descrição da mesma, para saber do que se trata. 
Ademais, não apenas o título e a descrição da issue devem seguir esse padrão, como também os comentários. Um "ok, bom trabalho" pode ser "recompensante", mas enche a issue de informação desnecessária. Agregar valor à moral da equipe pode ser feito pessoalmente ou em reuniões, mas as issues devem ser diretar e claras em informações e discussão.
